packages/drawer/src/views/DrawerItemList.tsx

```diff
@@ -27,6 +27,24 @@ export default function DrawerItemList({
 
   return (state.routes.map((route, i) => {
     const focused = i === state.index;
+
+    const onPress = () => {
+      const event = navigation.emit({
+        type: 'drawerPress',
+        target: route.key,
+        canPreventDefault: true,
+      });
+
+      if (!focused && !event.defaultPrevented) {
```

Comment:
Adding the `focused` check here is problematic. We still need to dispatch an action to close the drawer even when the screen is focused.

```suggestion
      if (!event.defaultPrevented) {
```